src/InteractiveWindow/Editor/InteractiveWindow.UIThreadOnly.cs

```diff
@@ -1378,6 +1403,11 @@ private void RemoveLastInputPrompt()
             private void AddLanguageBuffer()
             {
                 ITextBuffer buffer = _factory.CreateAndActivateBuffer(_window);
+                if (CurrentLanguageBuffer != null)
+                {
+                    _textBufferUndoManagerProvider.RemoveTextBufferUndoManager(CurrentLanguageBuffer);
```

Comment:
Per our debugging session, this is already being cleaned up by CreateAndActivateBuffer since it calls IVsPersistDocData.Close() which under the covers unregisters it for us.